cmd/config.go

```diff
@@ -171,6 +174,71 @@ func readEnvConfig() (conf Config, err error) {
 	return conf, nil
 }
 
+// This checks for conflicting options and turns any shortcut options (i.e. duration, iterations,
+// stages) into the proper scheduler configuration
+func buildExecutionConfig(conf Config) (Config, error) {
+	result := conf
+	if conf.Duration.Valid {
+		if conf.Iterations.Valid {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and iterations is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Stages != nil {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and stages is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Execution != nil {
+			return result, errors.New("specifying both duration and execution is not supported")
+		}
+
+		ds := scheduler.NewConstantLoopingVUsConfig(lib.DefaultSchedulerName)
+		ds.VUs = conf.VUs
+		ds.Duration = conf.Duration
+		result.Execution = scheduler.ConfigMap{lib.DefaultSchedulerName: ds}
+	} else if conf.Stages != nil {
+		if conf.Iterations.Valid {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both iterations and stages is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Execution != nil {
+			return conf, errors.New("specifying both stages and execution is not supported")
+		}
+
+		ds := scheduler.NewVariableLoopingVUsConfig(lib.DefaultSchedulerName)
+		ds.StartVUs = conf.VUs
+		for _, s := range conf.Stages {
+			if s.Duration.Valid {
+				ds.Stages = append(ds.Stages, scheduler.Stage{Duration: s.Duration, Target: s.Target})
+			}
+		}
+		result.Execution = scheduler.ConfigMap{lib.DefaultSchedulerName: ds}
+	} else if conf.Iterations.Valid || conf.Execution == nil {
+		// Either shared iterations were explicitly specified via the shortcut option, or no execution
+		// parameters were specified in any way, which will run the default 1 iteration in 1 VU
+		if conf.Iterations.Valid && conf.Execution != nil {
+			return conf, errors.New("specifying both iterations and execution is not supported")
+		}
+
+		ds := scheduler.NewSharedIterationsConfig(lib.DefaultSchedulerName)
+		ds.VUs = conf.VUs
+		if conf.Iterations.Valid { // TODO: fix where the default iterations value is set... sigh...
+			ds.Iterations = conf.Iterations
+		}
```

Comment:
Shouldn't the `if` be removed ? Whether the iterations was set or not we want to set the scheduler to the value, no matter whether it was provided by the user or it is the default value. And I prefer if it comes from the same place in all cases :)

Reply:
See the `TODO` :smile: 
Unfortunately, we can't do what you (and I) want until we fix the [configuration mess](https://github.com/loadimpact/k6/issues/883). The default number of `1` iterations comes not from the [CLI config's default value](https://github.com/loadimpact/k6/blob/8bcf39ac2d5d4c4b8d42dda25dedfe5b7fe3a64f/cmd/options.go#L50), where other default values are set, even if that is problematic in and of itself. Instead, it is [set in the code for the `k6 run` command](https://github.com/loadimpact/k6/blob/8bcf39ac2d5d4c4b8d42dda25dedfe5b7fe3a64f/cmd/run.go#L153), after `getConsolidatedConfig()` (and thus `buildExecutionConfig()`) is called. So this `if` here is meant to preserve the default `1` value set for `iterations` in `scheduler.NewSharedIterationsConfig()` if the user doesn't specify anything else...

Reply:
This was accidentally resolved by https://github.com/loadimpact/k6/pull/913/commits/507c5c5142639742b30e1d1ac5e492b21f0270d5
Not that there aren't other issues that are caused by the root cause of this one...